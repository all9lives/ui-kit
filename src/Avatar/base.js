import { css } from 'styled-components'
import React, { Component } from 'react'
import Unit from '../utils/Unit'
import Size from '../utils/Size'
import Color from '../utils/Color'
import View from '../View'
import Image from '../Image'
import Text from '../Text'

const SIZE_MAPPING = {
  small: 2,
  regular: 3,
  large: 4
}

export const Label = View.extend`
  position: absolute;
  border-radius: ${Unit.parse(3)};
  padding: ${Unit.parse(Size.spacing * 0.3)};
  background-color: ${Color.primary_light};
  right: ${Unit.parse(Size.spacing * 0.1)};
  bottom: ${props => props.name ? Unit.parse(Size.spacing * 2) : Unit.parse(Size.spacing * 0.1)};
`

export const LabelText = Text.extend`
  font-size: ${Unit.parse(Size.small)};
  color: ${Color.white};
`

export const Name = View.extend`
  margin-top: ${Unit.parse(Size.spacing * 0.5)};
  background-color: ${Color.gray};
  border-radius: ${Unit.parse(Size.small)};
  align-items: center;
  justify-content: center;
`

export const NameText = Text.extend`
  font-size: ${Unit.parse(Size.small)};
  color: ${Color.white};
`

export class Base extends Component {
  static defaultProps = {
    source: require('../assets/avatar-cat.jpg')
  }
  renderLabel = ({ label, name, labelStyle, labelTextStyle }) => (
    <Label name={name} style={labelStyle}>
      <LabelText style={labelTextStyle}>{label}</LabelText>
    </Label>
  )
  renderName = ({ name, nameStyle, nameTextStyle }) => (
    <Name style={nameStyle}>
      <NameText style={nameTextStyle}>{name}</NameText>
    </Name>
  )
  render () {
    const { size, source, name, label, borderRadius } = this.props
    return (
      <View direction='column' {...this.props}>
        <Image size={Size.icon * SIZE_MAPPING[size]} source={source} style={{ borderRadius }} />
        {name && this.renderName(this.props)}
        {label && this.renderLabel(this.props)}
      </View>
    )
  }
}

const base = css`
  position: relative;
`

export default base
