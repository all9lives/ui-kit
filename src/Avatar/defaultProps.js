export default {
  size: 'regular',
  borderRadius: 0
}
