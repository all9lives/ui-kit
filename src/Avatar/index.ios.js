import Avatar from './index.android'

export default Avatar
