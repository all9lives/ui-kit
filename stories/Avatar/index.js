import React from 'react'
import createStory from '../../storybook/createStory'
import Avatar from '../../src/Avatar'

createStory('Avatar', module)
  .add('default', () => <Avatar />)
  .add('borderRadius:10', () => <Avatar borderRadius={10} />)
  .add('name:Jerry', () => <Avatar name='Jerry' />)
  .add('nameStyle', () => <Avatar name='Jerry' nameStyle={{ backgroundColor: 'red' }} />)
  .add('nameTextStyle', () => <Avatar name='Jerry' nameTextStyle={{ color: 'red' }} />)
  .add('label:Cool', () => <Avatar label='Cool' />)
  .add('labelStyle', () => <Avatar label='Cool' labelStyle={{ backgroundColor: 'black' }} />)
  .add('labelTextStyle', () => <Avatar label='Cool' labelTextStyle={{ color: 'red' }} />)
  .add('name:Jerry,label:Cool', () => <Avatar name='Jerry' label='Cool' />)
  .add('size:small', () => <Avatar size='small' />)
  .add('size:large', () => <Avatar size='large' />)
  .add('resizeMode:cover', () => <Avatar resizeMode='cover' />)
