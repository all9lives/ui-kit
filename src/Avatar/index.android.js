import styled from 'styled-components/native'
import base, { Base } from './base'
import defaultProps from './defaultProps'

const Avatar = styled(Base)`
  ${base}
`
Avatar.defaultProps = defaultProps

export default Avatar
