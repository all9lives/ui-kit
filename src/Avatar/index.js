import styled from 'styled-components'
import base, { Base } from './base'
import defaultProps from './defaultProps'

const Avatar = styled(Base)`
  ${base}
`
Avatar.defaultProps = defaultProps

export default Avatar
