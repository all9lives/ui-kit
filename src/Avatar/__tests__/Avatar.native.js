/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Avatar from '../index.ios.js'

describe('Avatar.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Avatar>Apple</Avatar>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `borderRadius:5`', () => {
    const wrapper = shallow(
      <Avatar borderRadius={5} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `name:foo`', () => {
    const wrapper = shallow(
      <Avatar name='foo' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `label:boo`', () => {
    const wrapper = shallow(
      <Avatar label='boo' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `labelStyle`', () => {
    const wrapper = shallow(
      <Avatar labelStyle={{ backgroundColor: 'white' }} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `labelTextStyle`', () => {
    const wrapper = shallow(
      <Avatar labelTextStyle={{ color: 'white' }} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `nameStyle`', () => {
    const wrapper = shallow(
      <Avatar nameStyle={{ backgroundColor: 'white' }} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `nameTextStyle`', () => {
    const wrapper = shallow(
      <Avatar nameTextStyle={{ color: 'white' }} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `size:small`', () => {
    const wrapper = shallow(
      <Avatar size='small' />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
