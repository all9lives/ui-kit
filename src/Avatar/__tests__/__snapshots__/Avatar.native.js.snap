// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar.native renders a snapshot 1`] = `
<Base
  borderRadius={0}
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  Apple
</Base>
`;

exports[`Avatar.native renders a snapshot with \`borderRadius:5\` 1`] = `
<Base
  borderRadius={5}
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;

exports[`Avatar.native renders a snapshot with \`label:boo\` 1`] = `
<Base
  borderRadius={0}
  label="boo"
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;

exports[`Avatar.native renders a snapshot with \`labelStyle\` 1`] = `
<Base
  borderRadius={0}
  labelStyle={
    Object {
      "backgroundColor": "white",
    }
  }
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;

exports[`Avatar.native renders a snapshot with \`labelTextStyle\` 1`] = `
<Base
  borderRadius={0}
  labelTextStyle={
    Object {
      "color": "white",
    }
  }
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;

exports[`Avatar.native renders a snapshot with \`name:foo\` 1`] = `
<Base
  borderRadius={0}
  name="foo"
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;

exports[`Avatar.native renders a snapshot with \`nameStyle\` 1`] = `
<Base
  borderRadius={0}
  nameStyle={
    Object {
      "backgroundColor": "white",
    }
  }
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;

exports[`Avatar.native renders a snapshot with \`nameTextStyle\` 1`] = `
<Base
  borderRadius={0}
  nameTextStyle={
    Object {
      "color": "white",
    }
  }
  size="regular"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;

exports[`Avatar.native renders a snapshot with \`size:small\` 1`] = `
<Base
  borderRadius={0}
  size="small"
  source={1}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;
